### PR TITLE
Display metrics for ignored published events in Grafana [Component]

### PR DESCRIPTION
**Pull request by montaro**
See also #3627

### CODE REVIEW

**Comment by montaro** [comment]
image

**Comment by montaro** [comment]
image

File: components/event-bus/cmd/event-bus-publish-knative/publisher/publisher.go [hunk 1]
```diff
@@ -61,6 +63,13 @@ func (publisher *DefaultKnativePublisher) Publish(knativeLib *knative.KnativeLib
 	channel, err := knativeLib.GetChannel(*channelName, *namespace)
 	if err != nil {
 		log.Printf("cannot find the knative channel '%v' in namespace '%v'", *channelName, *namespace)
+		log.Println("incrementing total published messages counter")
+		log.Println("incrementing ignored messages counter")
```
Comment (marcobebway) [inline components/event-bus/cmd/event-bus-publish-knative/publisher/publisher.go]: Please consider combining the two logs together to have the full context in one log line.

File: components/event-bus/cmd/event-bus-publish-knative/metrics/metrics_def.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	Status           = "status"
+	SourceID         = "source_id"
+	EventType        = "event_type"
+	EventTypeVersion = "event_type_version"
+)
+
+var (
+	TotalPublishedMessages = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "total_published_messages",
+		Help: "The total number of ignored published messages",
```
[the remark this answers is not in the document]
Comment (montaro) [reply components/event-bus/cmd/event-bus-publish-knative/metrics/metrics_def.go]: Yeah this was missed after refactor by mistake

File: components/event-bus/cmd/event-bus-publish-knative/publisher/publisher.go (hunk 1 above)
Comment (montaro) [reply components/event-bus/cmd/event-bus-publish-knative/publisher/publisher.go]: Right, this was missed in the debugging phase

**Pull request merged**